Tasks/VsTest/VSTest.ps1

```diff
@@ -38,9 +43,17 @@ if($testAssemblyFiles)
 {
     Write-Verbose "Calling Invoke-VSTest for all test assemblies"
     $timeline = Start-Timeline -Context $distributedTaskContext
-	$cwd = Get-Location  
+    $projectName = Get-Variable -Context $distributedTaskContext -Name "System.TeamProject"
+    $buildDir = Get-Variable -Context $distributedTaskContext -Name "Agent.BuildDirectory" -Global $FALSE
+    $buildNumber = Get-Variable -Context $distributedTaskContext -Name "Build.BuildNumber"
+    $buildUri = Get-Variable -Context $distributedTaskContext -Name "Build.BuildUri"	
+    $cwd = $buildDir+"\"+$projectName
+    $testResultsDir = $buildDir+"\"+$projectName+"\"+"TestResults"
+    $trxLocation = $buildDir+"\"+$projectName+"\"+"Temp"
     Write-Verbose "Calling Invoke-VSTest from working folder: $cwd"
-    Invoke-VSTest -TestAssemblies $testAssemblyFiles -Timeline $timeline -VSTestVersion $vsTestVersion -TestFiltercriteria $testFiltercriteria -RunSettingsFile $runSettingsFile -PathtoCustomTestAdapters $pathtoCustomTestAdapters -CodeCoverageEnabled $codeCoverage -OtherConsoleOptions $otherConsoleOptions -WorkingFolder $cwd
+    Invoke-VSTest -TestAssemblies $testAssemblyFiles -Timeline $timeline -VSTestVersion $vsTestVersion -TestFiltercriteria $testFiltercriteria -RunSettingsFile $runSettingsFile -PathtoCustomTestAdapters $pathtoCustomTestAdapters -CodeCoverageEnabled $codeCoverage -OverrideTestrunParameters $overrideTestrunParameters -OtherConsoleOptions $otherConsoleOptions -WorkingFolder $cwd -TestResultsDir $testResultsDir -ResultLocation $trxLocation
```

Comment:
Are you even populating the timeline anymore?  I don't believe so and if not we should remove it.